horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/RunServiceImpl.java

```diff
@@ -1004,4 +1012,213 @@ public DatasetList listDatasetsBySchema(String uri, Integer limit, Integer page,
+                        String name = (String)obj[0];
+                        DoubleNode value = (DoubleNode) obj[1];
+                        root.set(name, value);
+                        matched = true;
+                     }
+
+                     results = unchecked(em.createNamedQuery(QUERY_1ST_LEVEL_ARRAY_BY_RUNID_TRANSFORMERID_SCHEMA_ID)
+                           .setParameter(1, run.id).setParameter(2, s.schemaid).setParameter(3, t.id)
+                           .unwrap(NativeQuery.class)
+                           .addScalar("name", TextType.INSTANCE)
+                           .addScalar("values", JsonNodeBinaryType.INSTANCE)
+                           .getResultList() );
+
+                     for (Object[] obj: results) {
+                        String name = (String)obj[0];
+                        ArrayNode value = (ArrayNode)obj[1];
+                        root.set(name, value);
+                        matched = true;
+                     }
+
+                     results = unchecked(em.createNamedQuery(QUERY_2ND_LEVEL_SCALAR_BY_RUNID_TRANSFORMERID_SCHEMA_ID) 
+                           .setParameter(1, run.id).setParameter(2, run.id).setParameter(3, s.schemaid).setParameter(4, t.id)
+                           .unwrap(NativeQuery.class)
+                           .addScalar("name", TextType.INSTANCE)
+                           .addScalar("value", JsonNodeBinaryType.INSTANCE)
+                           .getResultList() );
+
+                     for (Object[] obj: results) {
+                        String name = (String)obj[0];
+                        DoubleNode value = (DoubleNode)obj[1];
+                        root.set(name, value);
+                        matched = true;
+                     }
+
+                     results = unchecked(em.createNamedQuery(QUERY_2ND_LEVEL_ARRAY_BY_RUNID_TRANSFORMERID_SCHEMA_ID)
+                           .setParameter(1, run.id).setParameter(2, run.id).setParameter(3, s.schemaid).setParameter(4, t.id)
+                           .unwrap(NativeQuery.class)
+                           .addScalar("name", TextType.INSTANCE)
+                           .addScalar("values", JsonNodeBinaryType.INSTANCE)
+                           .getResultList() );
+
+                     for (Object[] obj: results) {
+                        String name = (String)obj[0];
+                        ArrayNode value = (ArrayNode)obj[1];
+                        root.set(name, value);
+                        matched = true;
+                     }
+
+                     extractorObjectsPerTrans.put(t, root);
+                  }
+
+                  boolean containsArray = false;
+                  int arrayPosition = -1;
+
+                  for (Transformer t : transformers) {
+                     ObjectNode objects = extractorObjectsPerTrans.get(t);
+                     if (t.function == null || t.function.isBlank()) {
+                        dataSets.add(new DataSet(run.start, run.stop
```

Comment:
I guess you want to use `transformedSingleNodeResults` here.

Reply:
I re-jigged things to avoid using a collection here to hold extractor results.